### PR TITLE
Makes planning actions sticky

### LINKED ISSUE #377: Makes planning actions sticky
**Is your feature request related to a problem? Please describe.**
Les boutons de modification d'une cellule sont actuellement bloquées en haut de la page ce qui nous oblige à faire des allers-retours entre la ligne à modifier et les boutons, notamment lorsque l'on souhaite modifier plusieurs utilisateurs sur des actions différentes.

**Describe the solution you'd like**
Il faudrait que les boutons d'actions soit flottants de façon à les rendre toujours facilement accessible qu'elle que soit la position du défilement dans la page.

### PR DESCRIPTION
**Pull request by laurentChin**
Closes #377

### CODE REVIEW

File: assets/css/availability-table.scss [hunk 1]
```diff
@@ -75,3 +75,15 @@ table.availability-table {
     color: white;
   }
 }
+
+@media screen and (min-width: 1001px) {
```
Comment (mRoca) [inline assets/css/availability-table.scss:79]: Why are you update the user availability table, as this PR is related to the planning page ?

File: assets/css/planning.scss [hunk 2]
```diff
@@ -169,3 +169,15 @@ $tableBoxSize: 40px;
     }
   }
 }
+
+@media screen and (min-width: 1001px) {
```
Comment (mRoca) [inline assets/css/planning.scss:173]: Those values should be the default one, and you should only use media query for the mobile versions

File: assets/css/planning.scss [hunk 3]
```diff
@@ -169,3 +169,15 @@ $tableBoxSize: 40px;
...
+  .planning-actions-container {
+    position: sticky;
+    top: 0;
```
Comment (mRoca) [inline assets/css/planning.scss:176]: Why are you hiding all the table sticky headers ?

File: assets/css/planning.scss [hunk 4]
```diff
@@ -169,3 +169,15 @@ $tableBoxSize: 40px;
...
+    z-index: 1;
+    width: 100vw;
```
Comment (mRoca) [inline assets/css/planning.scss:178]: you must put a size to the buttons container, as this line will create very large buttons (you can have a look to the screenshot I've put on your issue)

**Review by mRoca — CHANGES_REQUESTED** [review]

File: assets/css/planning.scss (hunk 3 above)
Comment (laurentChin) [reply assets/css/planning.scss:176]: thoses rules works with the rules in `assets/css/availability-table.scss`, to have both form actions and table headers sticky

File: assets/css/planning.scss (hunk 4 above)
Comment (laurentChin) [reply assets/css/planning.scss:178]: this width is not required for this PR, it's just to have the form actions container beeing not "floating" into the page but that's a cosmetic issue, the real need is to have the actions constantly accessible.

Comment (mRoca) [reply assets/css/planning.scss:178]: Please have a look to the screenshot I've put in the issue: the table headers are hidden, and buttons are now too large

Comment (laurentChin) [reply assets/css/planning.scss:178]: Yes I've seen the screenshot the PR is just not up to date.

File: assets/css/availability-table.scss (hunk 1 above)
Comment (laurentChin) [reply assets/css/availability-table.scss:79]: My bad, I put the rules in the wrong place, fix in in progress.

Comment (laurentChin) [reply assets/css/availability-table.scss]: Those rules are now removed, all the rules needed are in planning.scss.

File: assets/css/planning.scss (hunk 4 above)
Comment (laurentChin) [reply assets/css/planning.scss]: The table headers are fixed, Chrome doesn't apply the shift on `top` value on `thead` it has to be set on `th` cells.
I have add a `.planning-actions-container-wrapper`div to keep a background on the actions and prevent them to be "floating" in the page without removing the bootstrap sizing on action buttons.

**Comment by mRoca** [comment]
image

It would be nice to force the div to be sticky on the left as well

**Comment by mRoca** [comment]
I've a new issue on Chrome: the div is now fixed on the left, but if I only display 2 or 3 days, I can scroll on the right event if the page is fully displayed. When "scrolling" on the right, the page and and the horizontal scrollbar are growing at the same time.

It's weird, I can make a video if you need it to understand the problem

**Comment by laurentChin** [comment]
> I've a new issue on Chrome: the div is now fixed on the left, but if I only display 2 or 3 days, I can scroll on the right event if the page is fully displayed. When "scrolling" on the right, the page and and the horizontal scrollbar are growing at the same time.
> 
> It's weird, I can make a video if you need it to understand the problem

I did not succeed to reproduce this bug but I've had a constraint to prevent the floating behavior to be triggered if the table width is smaller than the window, I should've done this at the beginning.
Tell me if the bug is still here despite this constraint.

File: assets/css/planning.scss [hunk 5]
```diff
@@ -2,8 +2,18 @@
 
 $tableBoxSize: 40px;
 
+.planning-actions-container-wrapper {
+  width: 100vw;
```
Comment (mRoca) [inline assets/css/planning.scss:6]: This line forces the page to always have an horizontal scrollbar event with only one displayed day

**Comment by mRoca** [comment]
There is still the bug, but with bigger screen. In order to reproduce:
- On Chrome (Chromium 79 on my computer)
- Go to the planning
- Display like 4 or 5 days
- Scroll to the right border with your touchpad (not the mouse)
- Scroll again to the right, you will see the window growing

=> I can show you on Discord, by sharing my screen

**Comment by laurentChin** [comment]
That's ok I can reproduce it on Chrome Windows with Browserstack, on Mac OS the scroll is handled differently.

**Comment by mRoca** [comment]
Hi @laurentChin , any update?

**Comment by laurentChin** [comment]
Hi @mRoca, no update at this point I’m not satisfied with any of my fixes, they are over-complicated for the behavior I expect, so I’m gonna rewrite the initial implementation to make it more elegant. I just need to find some time to work on it.

**Pull request closed without merging**